src/io/iohk/decco/auto/instances/NativeInstances.scala

```diff
@@ -10,6 +10,19 @@ trait NativeInstances {
   private def bool2byte(b: Boolean): Byte = if (b) 1 else 0
   private def byte2bool(b: Byte): Boolean = b != 0
 
+  private def toArray(byteBuffer: ByteBuffer): Array[Byte] = {
+    if (byteBuffer.hasArray)
+      byteBuffer.array
+    else {
+      (byteBuffer: java.nio.Buffer).position(0)
+      val arr = new Array[Byte](byteBuffer.remaining())
+      byteBuffer.get(arr)
+      arr
```

Comment:
Should you reset the byte  buffer position?

Reply:
No, the position wasn't on 0 precisely because how the codecs work, but in reality it seems to be better to put it there.